src/main/java/com/pokegoapi/api/map/pokemon/EncounterResult.java

```diff
@@ -17,38 +17,108 @@
 
 import POGOProtos.Data.Capture.CaptureProbabilityOuterClass;
 import POGOProtos.Map.Pokemon.WildPokemonOuterClass;
+import POGOProtos.Networking.Responses.DiskEncounterResponseOuterClass;
 import POGOProtos.Networking.Responses.EncounterResponseOuterClass.EncounterResponse;
+import org.omg.PortableInterceptor.SUCCESSFUL;
 
 
-public class EncounterResult {
+public abstract class EncounterResult {
 
-	private EncounterResponse response;
 
-	public EncounterResult(EncounterResponse response) {
-		this.response = response;
-	}
+	public abstract boolean wasSuccessful();
 
-	public EncounterResponse.Status getStatus() {
-		return response == null ? null : response.getStatus();
-	}
+	/**
+	 * Return the status of the encounter
+	 *
+	 * @return status of results
+	 */
+	public abstract EncounterResponse.Status getStatus();
 
-	public boolean wasSuccessful() {
-		return response != null && getStatus() != null && getStatus().equals(EncounterResponse.Status.ENCOUNTER_SUCCESS);
-	}
+	public static class DiskEncounterResult extends EncounterResult {
```

Comment:
not a big fan of inner as well, but in this case the whole class it's 124 line, it won't be too hard to navigate thru that.

if there will be need for other EncounterResponse (tm) i will split it in different classes